hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azurebfs/AzureBlobFileSystemStore.java

```diff
@@ -1438,8 +1436,29 @@ public void rename(final Path source, final Path destination,
       /*
        * Fetch the list of blobs in the given sourcePath.
        */
-      List<BlobProperty> srcBlobProperties = getListBlobs(source, null,
-          tracingContext, null, true);
+      StringBuilder listSrcBuilder = new StringBuilder(
+          source.toUri().getPath());
+      if (!source.isRoot()) {
+        listSrcBuilder.append(FORWARD_SLASH);
+      }
+      String listSrc = listSrcBuilder.toString();
+      BlobList blobList = client.getListBlobs(null, listSrc, null,
+              tracingContext).getResult()
+          .getBlobList();
+      String nextMarker = blobList.getNextMarker();
+      List<BlobProperty> srcBlobProperties = blobList.getBlobPropertyList();
+
+      ListBlobQueue listBlobQueue = new ListBlobQueue(
```

Comment:
Queue doesn't take any thread. But does add to memory. Though it would be GCed since it would not be used anywhere else.
Agree that its bad. Fixing this.

Commit https://github.com/ABFSDriver/AbfsHadoop/pull/52/commits/c9f9c29e3d8b415c95bd60b75956906ec72bb654 has the fix.
We now create queue only either list in src has size >0 or it is an empty directory.